ipaserver/install/certs.py

```diff
@@ -659,6 +660,38 @@ def request_service_cert(self, nickname, principal, host):
                                              subject=host,
                                              passwd_fname=self.passwd_fname)
 
+    def cert_issued_by_ipa(self, nickname):
+        """
+        Return True if the certificate contained in the CertDB with the
+        provided nickname has been issued by IPA (either directly or by a
+        lighweight subca).
+
+        Note that this method can only be executed if api has been initialized
+        """
+        # This method needs to compare the subject name of the root CA
+        # from the NSS DB and from LDAP, because nicknames are not
+        # always aligned.
+
+        # From the NSSDB, retrieve the chain that issued the server cert
+        # (retrieve the whole chain because the cert may be
+        # issued by an IPA lighweight CA or IPA CA)
+        # and extract the root CA subject
+        try:
+            db_ca_chain = self.find_root_cert(nickname)[:-1]
+            db_ca_nickname = db_ca_chain[-1]
+            db_cacert = self.get_cert_from_db(db_ca_nickname)
+            db_cacert_subject = DN(db_cacert.subject)
+        except Exception:
```

Comment:
I'd rather avoid using bare `Exception`. Would `CalledProcessError` do here?
Also, IMO it would make sense to take either the first two or three steps and have them in a separated function (`get_issuer_db_nickname`/`get_issuer_cert_from_db`).